### PR TITLE
Kill DoubleVolumeDataLayout

### PR DESCRIPTION
**Pull request by achaikou**
This PR aims to remove DoubleVolumeDataLayout as I believe we should try avoid inheriting from openvds classes where possible. I think inheriting from their classes was not intended by openvds, thus it might cause troubles in future during updates.

### CODE REVIEW

File: testdata/scripts/make_well_known_depth_sample_axis.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import argparse
+from make_vds import *
```
Comment (yngve793) [inline testdata/scripts/make_well_known_depth_sample_axis.py:2]: Please specify what functions and classes that are used.
I think it is better to import segyio directly, instead of importing it via 'make_vds'

File: tests/gtest/datahandle_metadata_test.cpp [hunk 2]
```diff
@@ -327,4 +327,117 @@ TEST_F(DatahandleMetadataTest, Metadata_CRS_Double) {
...
+            WELL_KNOWN_DEFAULT_DATA.c_str(),
+            CREDENTIALS.c_str(),
+            CUSTOM_AXIS_ORDER_DATA.c_str(),
+            CREDENTIALS.c_str(),
+            binary_operator::SUBTRACTION
+        );
+
+        delete double_axis_order_datahandle;
+    },
+                testing::ThrowsMessage<std::runtime_error>(testing::HasSubstr("Dimension name mismatch for dimension 2: Inline versus Sample")));
```
Comment (yngve793) [inline tests/gtest/datahandle_metadata_test.cpp:403]: Preferably a mismatch axis order should give an axis order mismatch error. Wrong axis name is fundamentally a different issue.

**Review by yngve793 — COMMENTED** [review]
The PR does what it says on the tin and removes the DoubleVolumeDataLayout. Tests and code works as intended. Comments with suggested improvements are added.

File: tests/gtest/coordinate_transformer_test.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include <OpenVDS/OpenVDS.h>
+
+namespace {
+
+const std::string CREDENTIALS = "";
+
+TEST(DoubleCoordinateTransformerTest, A_Before_B) {
```
[the remark this answers is not in the document]
Comment (yngve793) [reply tests/gtest/coordinate_transformer_test.cpp:14]: Consider, regular_8x2_cube.vds and shift_4_8x2_cube.vds.

The intersecting cube will have the following axis definitions:
Inline [15, 18, 21, 24]
XLine [10, 12, 14, 16]
Sample [20, 24,28, ..., 128]

Let us define a position as [InLine, XLine, Sample]. 
Let [15, 10, 20] be the minimal voxel and  [24, 16, 128] the maximal voxel.

We can now test the edge cases. For the minimum voxel we get:
[15, 10, 20] Valid
[12, 10, 20] Invalid
[15, 8, 20] Invalid
[15, 10, 16] Invalid

For the maximum we get:
[24, 16, 128] Valid
[27, 16, 128] Invalid
[24, 18, 128] Invalid
[24, 16, 132] Invalid

**Review by yngve793 — COMMENTED** [review]
Closed most of the issues. Added more detailed comment for the remaining.

File: internal/core/coordinate_transformer.hpp [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    OpenVDS::IJKCoordinateTransformer coordinate_transformer;
+};
+
+class DoubleCoordinateTransformer : public CoordinateTransformer {
+public:
+    DoubleCoordinateTransformer(
+        SingleCoordinateTransformer const& transformer_a,
+        SingleCoordinateTransformer const& transformer_b
+    )
+        : m_transformer_a(transformer_a) {
```
[the remark this answers is not in the document]
Comment (achaikou) [reply internal/core/coordinate_transformer.hpp:58]: I don't think it was enforced :innocent:  There are arguments for splitting and arguments against that.
I think for the moment there is no that great need to split coordinate_transformer, seems like more clutter than the opposite. But I am 100% fine with doing that if needed.

**Review by yngve793 — APPROVED** [review]
No further comments. 

Due to the compile issue I am not able to run tests via rebase.
